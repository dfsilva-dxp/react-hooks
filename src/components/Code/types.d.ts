import { ReactNode } from "react";

export interface ICodeProps {
  children: ReactNode;
}
