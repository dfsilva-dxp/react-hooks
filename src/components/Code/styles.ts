import styled, { css } from "styled-components";

export const CodeWrapper = styled.code`
  ${({ theme }) => css`
    background-color: ${theme.colors.divider};
    padding: 2px ${theme.spacings.xxsmall};
    font-size: 0.94em;
    word-break: break-word;
    border-radius: 2px;
    color: ${theme.colors.green};
  `}
`;
