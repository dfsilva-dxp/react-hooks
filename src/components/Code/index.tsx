import { ICodeProps } from "./types";

import * as S from "./styles";

export default function Code({ children }: ICodeProps) {
  return (
    <S.CodeWrapper role="code" aria-label="Code">
      {children}
    </S.CodeWrapper>
  );
}
