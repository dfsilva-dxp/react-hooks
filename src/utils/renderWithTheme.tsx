import { ReactNode } from "react";
import { render, RenderResult } from "@testing-library/react";
import { ThemeProvider } from "styled-components";

import { MyTheme } from "../styles";

export const renderWithTheme = (children: ReactNode): RenderResult =>
  render(<ThemeProvider theme={MyTheme}>{children}</ThemeProvider>);
