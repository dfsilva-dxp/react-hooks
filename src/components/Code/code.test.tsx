import { screen } from "@testing-library/dom";
import { renderWithTheme } from "../../utils/tests/renderWithTheme";

import Code from ".";

describe("<Code/>", () => {
  it("should render currectly", () => {
    const { container } = renderWithTheme(<Code>HTML</Code>);

    expect(screen.getByRole("code")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
