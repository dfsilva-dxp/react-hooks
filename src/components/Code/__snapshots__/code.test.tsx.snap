// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Code/> should render currectly 1`] = `
.c0 {
  background-color: #323238;
  padding: 2px 0.5rem;
  font-size: 0.94em;
  word-break: break-word;
  border-radius: 2px;
  color: #00B37E;
}

<div>
  <code
    aria-label="Code"
    class="c0"
    role="code"
  >
    HTML
  </code>
</div>
`;
